### PR TITLE
🪟 Allow users to create new connection regardless of credit balance

### LINKED ISSUE #12483: Allow users to create new connection regardless of credit balance
Followup to #11775 - Update the button logic to allow cloud users to create new connections even if the workspace ran out of credits. This wasn't possible before until #12717 was merged.

### Context

When creating a new connection, the button is sometimes greyed out and not clickable. It seems to be an issue when a source or destination has been set up. (example workspace: https://cloud.airbyte.io/workspaces/80c785cb-bdd9-4034-9414-8763eeff1188/connections)
Screen Shot 2022-04-29 at 5 21 37 PM

### PR DESCRIPTION
**Pull request by dizel852**
## What
Closes #12483 

## How
Enable "new connection" buttons even if the credit balance is not positive

image
image

## Recommended reading order
All

### CODE REVIEW

**Comment by nataliekwong** [comment]
@dizel852 No, since we enabled the buttons, we don't need to show any message.

I mentioned destinations as an example (agreed it doesn't make sense for a add connection button) for text in case we grey the button out. But we don't need this given we enable the button now.

**Comment by dizel852** [comment]
> @dizel852 No, since we enabled the buttons, we don't need to show any message.
> 
> I mentioned destinations as an example (agreed it doesn't make sense for a add connection button) for text in case we grey the button out. But we don't need this given we enable the button now.

Oh, okay. Now clear 😄

File: airbyte-webapp/src/pages/ConnectionPage/pages/AllConnectionsPage/AllConnectionsPage.tsx [hunk 1]
```diff
@@ -41,11 +39,7 @@ const AllConnectionsPage: React.FC = () => {
           <ConnectionsTable connections={connections} />
         </MainPageWithScroll>
       ) : (
-        <EmptyResourceListView
-          resourceType="connections"
-          onCreateClick={onCreateClick}
-          disableCreateButton={!allowCreateConnection}
-        />
+        <EmptyResourceListView resourceType="connections" onCreateClick={onCreateClick} />
```
Comment (krishnaglick) [inline airbyte-webapp/src/pages/ConnectionPage/pages/AllConnectionsPage/AllConnectionsPage.tsx:42]: You can remove `disableCreateButton` from this component now.

**Review by krishnaglick — COMMENTED** [review]
Based on the conversation above, should the Destination page prevent creation with a 0 balance?
Looking for a good way to test this locally.

File: airbyte-webapp/src/pages/ConnectionPage/pages/AllConnectionsPage/AllConnectionsPage.tsx (hunk 1 above)
Comment (dizel852) [reply airbyte-webapp/src/pages/ConnectionPage/pages/AllConnectionsPage/AllConnectionsPage.tsx:42]: Fixed

**Comment by dizel852** [comment]
> Based on the conversation above, should the Destination page prevent creation with a 0 balance?
> Looking for a good way to test this locally.

Hmm... what I understand from #12483 is that we need to allow users to create sources/destinations/connections, but not run sync jobs...

@nataliekwong please confirm

**Review by krishnaglick — APPROVED** [review]
LGTM pending Natalie

**Comment by nataliekwong** [comment]
> Hmm... what I understand from https://github.com/airbytehq/airbyte/issues/12483 is that we need to allow users to create sources/destinations/connections, but not run sync jobs...

Yes, that's correct - we would allow them to do all the creation actions (source, destination, connection) except run syncs.

**Comment by timroes** [comment]
@dizel852 Late feedback, sorry: I think there is one place missing from this PR: we still have the "add destination" or "add source" button on the detail page of a source/destination to create a connection that still seems to follow the check: https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/ConnectorBlocks/TableItemTitle.tsx#L79

Also if we remove this we don't need the `FeatureItem.AllowCreateConnection` anymore at all, and should remove this altogether.

cc @edmundito

**Comment by nataliekwong** [comment]
Is there a timeline to resolve the missing part of this PR that Tim pointed out above? This was flagged by Support.

cc @edmundito @andyjih @salima-airbyte

**Comment by dizel852** [comment]
@nataliekwong I'm on it

**Comment by dizel852** [comment]
> PR: #18372

@nataliekwong merged ✅

**Pull request merged**